cmake/platform/linux/wayland.cmake

```diff
@@ -0,0 +1,17 @@
+set(PLATFORM_REQUIRED_DEPS EGL Waylandpp LibDRM Xkbcommon)
+set(PLATFORM_OPTIONAL_DEPS VAAPI)
+
+set(WAYLAND_RENDER_SYSTEM "" CACHE STRING "Render system to use with Wayland: \"gl\" or \"gles\"")
+
+if(WAYLAND_RENDER_SYSTEM STREQUAL "gl")
+  list(APPEND PLATFORM_REQUIRED_DEPS OpenGl)
+elseif(WAYLAND_RENDER_SYSTEM STREQUAL "gles")
+  list(APPEND PLATFORM_REQUIRED_DEPS OpenGLES)
+else()
+  message(SEND_ERROR "You need to decide whether you want to use GL- or GLES-based rendering in combination with the Wayland windowing system. Please set WAYLAND_RENDER_SYSTEM to either \"gl\" or \"gles\". For normal desktop systems, you will usually want to use \"gl\".")
```

Comment:
> I am pretty sure this will cause confusion. Having gl as default on x86 is hardly unpredictable

Hm, not sure why it would be confusing to have to choose. The message also explicitly states to try `gl` on normal systems (we can of course reword the message if it isn't clear enough). And if you're compiling for embedded you should know whether you want GLES or not.